### PR TITLE
Switch to AWS CI/CD

### PR DESCRIPTION
**Pull request by andreyvelich**
Related: https://github.com/kubeflow/katib/issues/1332.
I will debug the infra in this PR.

I also made few changes to improve CI/CD quality.

/cc @gaocegege @johnugeorge 
/cc @Jeffwan @PatrickXYS @jlewi @Bobgy

### CODE REVIEW

File: test/scripts/v1beta1/create-cluster.sh
Comment (PatrickXYS) [inline test/scripts/v1beta1/create-cluster.sh:37]: Can we use managed nodegroup? 

Check here: https://github.com/kubeflow/testing/blob/master/images/aws-scripts/create-eks-cluster.sh#L36

Comment (andreyvelich) [reply test/scripts/v1beta1/create-cluster.sh:37]: I think I will use your script to create EKS cluster: https://github.com/kubeflow/testing/blob/master/images/aws-scripts/create-eks-cluster.sh.
What do you think @PatrickXYS ?

**Comment by PatrickXYS** [comment]
Pasted from https://github.com/kubeflow/katib/pull/1357

The reason that plugin trigger can not trigger prow test:

```
{"author":"PatrickXYS","component":"hook","error":"error checking trust of PatrickXYS: error in IsCollaborator: return code not 2XX: 403 Forbidden","event-GUID":"16d24480-0fdd-11eb-911f-618d74ee83e5","event-type":"issue_comment","file":"prow/hook/events.go:425","func":"k8s.io/test-infra/prow/hook.(*Server).handleGenericComment.func1","level":"error","msg":"Error handling GenericCommentEvent.","org":"kubeflow","plugin":"trigger","pr":1357,"repo":"katib","severity":"error","time":"2020-10-16T18:26:29Z","url":"https://github.com/kubeflow/katib/pull/1357#issuecomment-710377544"}
```

It's caused by `IsCollaborrator` error, since `k8s-ci-bot` is admin of kubeflow org, so by-default it's `Collaborator` of any kubeflow repo.

**Comment by andreyvelich** [comment]
@PatrickXYS Should we remove access k8s-ci-bot to Katib repo?
Or in that case, tide also doesn't work?

**Comment by PatrickXYS** [comment]
Removing `kf-ci-bot` from katib might be un-acceptable by google folks.

As we discussed in the [proposal](https://docs.google.com/document/d/1AMz4sS_PF7ro5AFaEqFeJwtRUDswiPp5zbzyhpWlOe0/edit?disco=AAAAG_t1JRg), Google Prow owns Tide functionality, and AWS Prow owns running tests and report PR status.

This should be one of the potential issue with mixed Prow, I'll keep you updated with the progress

/cc @andreyvelich

**Comment by Jeffwan** [comment]
The problem is 

> @PatrickXYS: GitHub didn't allow me to request PR reviews from the following users: andreyvelich.
> 
> Note that only [kubeflow members](https://github.com/orgs/kubeflow/people) and repo collaborators can review this PR, and authors cannot review their own PRs.

em.. This should be unexpected @andreyvelich is definitely a member.  Anything wrong with the upstream prow because of new prow?

**Comment by PatrickXYS** [comment]
> and authors cannot review their own PRs.

@Jeffwan  This is expected, because @andreyvelich can't review his own PR.

**Comment by Jeffwan** [comment]
@PatrickXYS seems bot does have access to the repo. However, I didn't see it can add label. `approve` label is added by `k8s-ci-bit`. Can you check backend logs?

**Comment by Jeffwan** [comment]
> > and authors cannot review their own PRs.
> 
> @Jeffwan This is expected, because @andreyvelich can't review his own PR.

I see. my careless. I thought it's failed to assign.

**Comment by PatrickXYS** [comment]
> seems bot does have access to the repo. However, I didn't see it can add label. approve label is added by k8s-ci-bot. Can you check backend logs?

Yeah I checked AWS prow backend logs, and the `approve` label should be added by `k8s-ci-bot`.

**Comment by PatrickXYS** [comment]
So in a word, the reason AWS Prow can't trigger Presubmit is:

Prow Plugin `Trigger` failed to check if user is `IsCollaborator` when user send PR.

File: test/scripts/v1beta1/create-cluster.sh
Comment (PatrickXYS) [reply test/scripts/v1beta1/create-cluster.sh:37]: > I think I will use your script to create EKS cluster
Yes we can, reuse common script makes a lot of sense to me.

**Comment by PatrickXYS** [comment]
Woo here we go

**Comment by PatrickXYS** [comment]
http://8069b97e-prow-prow-6530-1439048706.us-west-2.elb.amazonaws.com/

-> Presubmit-katib-e2e is kicked off

**Comment by PatrickXYS** [comment]
The reason that `presubmit-katib-e2e` failed:

```
 Warning  FailedMount  42m (x61 over 16h)    kubelet  Unable to attach or mount volumes: unmounted volumes=[gcp-credentials], unattached volumes=[default-token-lc29r podmetadata docker-sock kubeflow-test-volume github-token gcp-credentials]: timed out waiting for the condition
  Warning  FailedMount  28m (x76 over 16h)    kubelet  Unable to attach or mount volumes: unmounted volumes=[gcp-credentials], unattached volumes=[github-token gcp-credentials default-token-lc29r podmetadata docker-sock kubeflow-test-volume]: timed out waiting for the condition
  Warning  FailedMount  13m (x480 over 16h)   kubelet  MountVolume.SetUp failed for volume "gcp-credentials" : secret "kubeflow-testing-credentials" not found
  Warning  FailedMount  8m27s (x63 over 16h)  kubelet  Unable to attach or mount volumes: unmounted volumes=[gcp-credentials], unattached volumes=[kubeflow-test-volume github-token gcp-credentials default-token-lc29r podmetadata docker-sock]: timed out waiting for the condition
  Warning  FailedMount  3m57s (x59 over 15h)  kubelet  Unable to attach or mount volumes: unmounted volumes=[gcp-credentials], unattached volumes=[gcp-credentials default-token-lc29r podmetadata docker-sock kubeflow-test-volume github-token]: timed out waiting for the condition
``` 

It still attach `gcp-credentials`.

Besides that, looks like katib argo workflow haven't set up timeout-duration for each stage of workflow, which makes the pod stuck there for a long time. 

@andreyvelich Can we set up similar configuration for argo workflow, you can check https://github.com/kubeflow/kfctl/blob/master/testing/workflows/components/unit_tests.jsonnet#L48

Given katib workflow needs more time to pass one stage, we can set up timeout-duration to be 2 hours.

**Comment by PatrickXYS** [comment]
The reason that Github Status can't be updated:

```
{"component":"crier","error":"error setting status: status code 404 not one of [201], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/repos#create-a-commit-status\"}","file":"prow/crier/controller.go:277","func":"k8s.io/test-infra/prow/crier.(*Controller).processNextItem","jobName":"presubmit-katib-e2e","jobStatus":"success","level":"error","msg":"failed to report job","prowjob":"prow/9626395c-10d7-11eb-8094-aa5d9bf7e56e","reporter":"github-reporter","severity":"error","time":"2020-10-18T00:20:26Z"}

{"component":"crier","error":"error setting status: status code 404 not one of [201], body: {\"message\":\"Not Found\",\"documentation_url\":\"https://docs.github.com/rest/reference/repos#create-a-commit-status\"}","file":"prow/crier/controller.go:149","func":"k8s.io/test-infra/prow/crier.(*Controller).retry","jobName":"presubmit-katib-e2e","jobStatus":"success","level":"error","msg":"Failed processing item, no more retries","prowjob":"prow/9626395c-10d7-11eb-8094-aa5d9bf7e56e","reporter":"github-reporter","severity":"error","time":"2020-10-18T00:20:26Z"}
```

**Comment by andreyvelich** [comment]
The test passed: http://8069b97e-prow-prow-6530-1439048706.us-west-2.elb.amazonaws.com/view/s3/aws-kubernetes-jenkins/pr-logs/pull/kubeflow_katib/1356/presubmit-katib-e2e/1318507401464254464.

@gaocegege @johnugeorge Can you take a look at the changes, please?

I deleted v1alpha3 workflow and tests, I don't think we need to leave this test since we release v1beta1 soon.

**Comment by PatrickXYS** [comment]
> We still need to discuss which registry we will use to public images. AWS doesn't support it.

Can you include me as well when you're talking about this task?

**Comment by andreyvelich** [comment]
> > We still need to discuss which registry we will use to public images. AWS doesn't support it.
> 
> Can you include me as well when you're talking about this task?

@PatrickXYS Can you join today's Kubeflow community meeting? 5:30pm PST.

**Comment by PatrickXYS** [comment]
I tried to debug Github Status Report issue, per this [doc](https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#create-a-commit-status):

```
Users with push access in a repository can create commit statuses for a given SHA.
```

I'm not sure if aws-kf-ci-bot has push access

**Comment by andreyvelich** [comment]
@PatrickXYS Should we modify `aws-kf-ci-bot` Collaborator access?

**Comment by PatrickXYS** [comment]
@andreyvelich I believe we need to do this, I'm already asking for some progress on PR https://github.com/kubeflow/internal-acls/pull/357

As long as I have temporary admin, I'll start the process immediately

**Comment by PatrickXYS** [comment]
woohoo UI works!!!!!

**Review by Jeffwan — APPROVED** [review]

**Comment by andreyvelich** [comment]
We will use https://hub.docker.com/u/kubeflowkatib as temporary container registry for Katib images.
/hold cancel

/cc @johnugeorge @gaocegege @jlewi

**Comment by andreyvelich** [comment]
@johnugeorge @gaocegege I was thinking about temporary solution for the registry.

Actually, it is easier to use just another `workflow-v1beta1-release` workflow during post-submit for us.
This workflow just pushes images to `kubeflow-images-public` and doesn't deploy any clusters.
In that case, we don't need to change website/manifests/katib repo's docs with temporary container registry.
After AWS release process implementation, we will switch registry.

What do you think?

**Comment by jlewi** [comment]
> Actually, it is easier to use just another workflow-v1beta1-release workflow during post-submit for us.
This workflow just pushes images to kubeflow-images-public

How will these jobs have access to kubeflow-images-public?

**Comment by andreyvelich** [comment]
@jlewi Using `kubeflow-ci` service account. This is just shor-term solution, until we implement new release process in AWS.

**Pull request merged**
